homework 04

xRay64 opened the pull request:
spring boot advanced configuration
добавил spring shell
перевел unit-тесты на springBootTest и доработал тесты

Commit:
homework 04
spring boot advanced configuration
добавил spring shell
перевел unit-тесты на springBootTest и доработал тесты

kataus commented on homework-02/src/main/java/ru/otus/homework02/helper/IOServiceImpl.java:
```diff
@@ -0,0 +1,59 @@
+
+import java.io.*;
+
+@Scope("prototype")
```
А почему тут выбран именно прототип? Какую пользу мы от него получаем?

xRay64 replied:
Реализовал через scope=prototype для того, что бы возможно было получать разные бины с разными inputStream, outputStream которые предоставляет IOStreamsProvider.
В текущей реализации приложения эта фича нигде не используется кроме юнит-тестов в UserInteractionServiceImplTest

Commit:
homework 04
spring boot advanced configuration
исправил scope, что помогло избавиться от лишнего класса
переделал unit-тесты
